src/node/time.rs

```diff
@@ -1,82 +1,223 @@
 use anyhow::anyhow;
-use std::sync::{Arc, RwLock};
+use std::collections::VecDeque;
+use std::sync::{Arc, RwLock, RwLockReadGuard, RwLockWriteGuard};
+
+/// Shared readable view on time.
+pub trait TimeRead {
+    /// Returns the last timestamp (in seconds) that has already been used.
+    fn last_timestamp(&self) -> u64;
+
+    /// Peek at what the next call to `next_timestamp` will return.
+    fn peek_next_timestamp(&self) -> u64;
+}
+
+/// Writeable view on time management. The owner of this view should be able to treat it as
+/// exclusive access to the underlying clock.
+pub trait TimeExclusive: TimeRead {
+    /// Returns the next unique timestamp in seconds.
+    ///
+    /// Subsequent calls to this method return monotonically increasing values.
+    fn next_timestamp(&mut self) -> u64;
```

Comment:
IMHO it's unclear from the method name that the method is mutable. Maybe name it `advance_timestamp`?